vm/devices/storage/disklayer_ram/src/lib.rs

```diff
@@ -31,41 +32,54 @@ use std::sync::atomic::AtomicU64;
 use std::sync::atomic::Ordering;
 use thiserror::Error;
 
+#[doc(hidden)]
+pub enum PreAttach {}
```

Comment:
Why all the fancy type system stuff? Can't we just have something like `RamDiskLayer { .. }` and `LazyRamDiskLayer {}` as separate types?